libs/math/tests/math/all_tests/fixed_point/fixed_point.cpp

```diff
@@ -786,39 +807,172 @@ TEST(FixedPointTest, Exponential_32_32)
   EXPECT_THROW(fp64::Exp(fp64::MAX_EXP + 1), std::overflow_error);
 
   // Negative values
-  EXPECT_NEAR(((double)fp64::Exp(-one) - std::exp(-1.0)) / std::exp(-1.0), 0,
-              (double)fp64::TOLERANCE);
-  EXPECT_NEAR(((double)fp64::Exp(-two) - std::exp(-2.0)) / std::exp(-2.0), 0,
-              (double)fp64::TOLERANCE);
+  EXPECT_NEAR(((double)fp64::Exp(-one) - std::exp(-(double)one)), 0, (double)fp64::TOLERANCE);
+  EXPECT_NEAR(((double)fp64::Exp(-two) - std::exp(-(double)two)), 0, (double)fp64::TOLERANCE);
 
   // This particular error produces more than 1e-6 error failing the test
-  EXPECT_NEAR(((double)fp64::Exp(-ten) - std::exp(-10.0)) / std::exp(-10.0), 0, 1.1e-6);
+  EXPECT_NEAR((double)fp64::Exp(-ten) - std::exp(-(double)ten), 0, (double)fp64::TOLERANCE);
   // The rest pass with fp64::TOLERANCE
-  EXPECT_NEAR(((double)fp64::Exp(-small) - std::exp(-0.0001)) / std::exp(-0.0001), 0,
-              (double)fp64::TOLERANCE);
-  EXPECT_NEAR(((double)fp64::Exp(-tiny) - std::exp(-double(tiny))) / std::exp(-double(tiny)), 0,
+  EXPECT_NEAR((double)fp64::Exp(-small) - std::exp(-(double)small), 0, (double)fp64::TOLERANCE);
+  EXPECT_NEAR((double)fp64::Exp(-tiny) - std::exp(-double(tiny)), 0, (double)fp64::TOLERANCE);
+  EXPECT_NEAR((double)fp64::Exp(fp64::MIN_EXP) - std::exp((double)fp64::MIN_EXP), 0,
               (double)fp64::TOLERANCE);
-  EXPECT_NEAR(((double)fp64::Exp(fp64::MIN_EXP) - std::exp((double)fp64::MIN_EXP)) /
-                  std::exp((double)fp64::MIN_EXP),
-              0, (double)fp64::TOLERANCE);
+
+  fp64   step{0.0001};
+  fp64   x{-10.0};
+  double max_error = 0, avg_error = 0;
+  size_t iterations = 0;
+  double tolerance  = (double)fp64::TOLERANCE;
+  for (; x < 5.0; x += step)
+  {
+    fp64   e     = fp64::Exp(x);
+    double r     = std::exp((double)x);
+    double delta = std::abs((double)e - r);
+    max_error    = std::max(max_error, delta);
+    avg_error += delta;
+    iterations++;
+  }
+  avg_error /= (double)iterations;
+  EXPECT_NEAR(max_error, 0.0, 10 * tolerance);
```

Comment:
I am not sure what you mean about highest value of x, it may be true for some monotonic functions, but not all, it depends on the approximation used I guess, eg for tan(x), max_error will really skyrocket around PI/2.